0-Parameterised_reports.R

```diff
@@ -15,6 +15,64 @@ updates %>%
   as.list() %>%
   list2env(globalenv())
 
+# Test Raw Data  ==============================================================
+
+raw_data <- Filter(function(x) is(x, "data.frame"), mget(ls()))
+
+expected_file_names <-  c("updates",
+                           "population-estimates",
+                           "population-projections",
+                           "nature-of-population-change",
+                           "births-by-sex",
+                           "standardised-birth-rates",
+                           "births-by-age-of-mother",
+                           "fertility-rates",
+                           "deaths-by-sex",
+                           "standardised-death-rates",
+                           "deaths-by-sex-by-age",
+                           "leading-causes-of-death",
+                           "migration",
+                           "net-migration",
+                           "net-migration-rates",
+                           "life-expectancy",
+                           "marriages",
+                           "civil-partnerships",
+                           "household-estimates",
+                           "household-projections",
+                           "dwelling",
+                           "dwellings-by-type",
+                           "dwellings-by-council-tax-band")
+
+check_expected_data_files <- function(raw_data) {   #Check for expected data sheets
+  if(identical(sort(names(raw_data)), sort(expected_file_names)) == F){
+    stop("Unexpected Sheet Names")
+  }
+}
+
+
+check_blank_data <- function(raw_data) { #Check for NA's in data
+  
+  for(df in seq_along(raw_data)){
+    if(any(is.na(raw_data[[df]]))){

```

Comment:
Yep, because I want it to check level 2 of the list where the data is but the names of the sheets are stored at level one, so when there are blanks, it only says the name of the sheet.